worlds/yoshisisland/__init__.py

```diff
@@ -266,25 +260,24 @@ def get_excluded_items(self) -> Set[str]:
 
     def create_item_with_correct_settings(self, player: int, name: str) -> Item:
         data = item_table[name]
-        classification = self.item_classifications[data.classification]
-        item = Item(name, classification, data.code, player)
+        item = Item(name, data.classification, data.code, player)
 
         if not item.advancement:
             return item
 
-        if name == 'Car Morph' and self.options.stage_logic.value != 0:
+        if name == 'Car Morph' and not self.options.stage_logic:
             item.classification = ItemClassification.useful
 
-        if name == 'Secret Lens' and (self.options.hidden_object_visibility.value >= 2 or self.options.stage_logic.value != 0):
+        if name == 'Secret Lens' and (self.options.hidden_object_visibility >= 2 or self.options.stage_logic != 0):
```

Comment:
Instead of
`self.options.stage_logic != 0`
You probably want just
`self.options.stage_logic`